### PR TITLE
Nullness - generated ToString() members should override object and return string WithoutNull

### PR DESCRIPTION
**Pull request by T-Gro**
- Overriding .ToString on an F# type will warn if it returns nullable string
- Calling .ToString on F# types with generated ToString (powered by sprintfn %A under the hood) is typechecked as not returning null

### CODE REVIEW

File: src/Compiler/Checking/ConstraintSolver.fs [hunk 1]
```diff
@@ -2132,6 +2132,8 @@ and MemberConstraintSolutionOfMethInfo css m minfo minst staticTyOpt =
     | FSMeth(_, ty, vref, _) ->  
        FSMethSln(ty, vref, minst, staticTyOpt)
 
+    | MethInfoWithModifiedReturnType(mi,_) -> MemberConstraintSolutionOfMethInfo css m mi minst staticTyOpt
```
Comment (vzarytovskii) [inline src/Compiler/Checking/ConstraintSolver.fs:2135]: Thanks, I hate it :|

File: src/Compiler/Checking/InfoReader.fs [hunk 2]
```diff
@@ -1191,7 +1191,7 @@ let GetXmlDocSigOfUnionCaseRef (ucref: UnionCaseRef) =
         ucref.UnionCase.XmlDocSig <- XmlDocSigOfUnionCase [tcref.CompiledRepresentationForNamedType.FullName; ucref.CaseName]
     Some (ccuFileName, ucref.UnionCase.XmlDocSig)
 
-let GetXmlDocSigOfMethInfo (infoReader: InfoReader)  m (minfo: MethInfo) = 
+let rec GetXmlDocSigOfMethInfo (infoReader: InfoReader)  m (minfo: MethInfo) =
```
Comment (vzarytovskii) [inline src/Compiler/Checking/InfoReader.fs:1194]: Can we slap tailcall attribute on it?

**Review by vzarytovskii — APPROVED** [review]
Pseudo method type is not my favourite, but it's easy to understand.

File: src/Compiler/Checking/InfoReader.fs (hunk 2 above)
Comment (T-Gro) [reply src/Compiler/Checking/InfoReader.fs:1195]: yes

File: src/Compiler/FSComp.txt [hunk 3]
```diff
@@ -1536,6 +1536,7 @@ tcPassingWithoutNullToValueOptionOfObj,"You can create 'ValueSome value' directl
 tcPassingWithoutNullToNonNullAP,"You can remove this |Null|NonNull| pattern usage."
 tcPassingWithoutNullToNonNullQuickAP,"You can remove this |NonNullQuick| pattern usage."
 tcPassingWithoutNullTononNullFunction,"You can remove this `nonNull` assertion."
+3263,tcNullableToStringOverride,"With nullness checking enabled, overrides of .ToString() method must return a non-nullable string. You can handle potential nulls via the built-in string function."
```
Comment (psfinaki) [inline src/Compiler/FSComp.txt:1539]: For clarity, would prefer
```suggestion
3263,tcNullableToStringOverride,"With nullness checking enabled, overrides of .ToString() method must return a non-nullable string. You can handle potential nulls via the built-in `string` function."
```

**Review by psfinaki — APPROVED** [review]

File: src/Compiler/FSComp.txt (hunk 3 above)
Comment (T-Gro) [reply src/Compiler/FSComp.txt:1539]: This does not look good without markdown, I have not seen it used elsewhere apart from the `nonNull` above.

**Pull request merged**